### DISCUSSION TITLE
v4.0.0

### DISCUSSION THREAD

**Discussion by jhuleatt** [FirebaseExtended/reactfire#428] [General]
ReactFire version 4 supports [Firebase version 9](https://firebase.googleblog.com/2021/08/the-new-firebase-js-sdk-now-ga.html). Firebase v9 introduces a new API that is more tree-shakeable, which should result in reduced bundle size and therefore faster page loads for your users!

[**ReactFire v3 -> v4 upgrade guide**](https://github.com/FirebaseExtended/reactfire/blob/main/docs/upgrade-guide.md)

Check out [discussion #402](https://github.com/FirebaseExtended/reactfire/discussions/402) for more context.

ReactFire v3 source is available in the [`v3` branch](https://github.com/FirebaseExtended/reactfire/tree/v3).

<hr /><em>This discussion was created from the release <a href='https://github.com/FirebaseExtended/reactfire/releases/tag/v4.0.0'>v4.0.0</a>.</em>

**Comment by jhuleatt** [comment]
The [*Using ReactFire* doc](https://github.com/FirebaseExtended/reactfire/blob/main/docs/use.md#using-reactfire) has been updated with setup instructions, how to connect to emulators, and a number of new examples.

**Reply by shiiinji** [reply]
Hi,

I used useAnalytics with reactfire 3.x.
But when I upgraded to reactfire 4.x, useAnalytics has no exported from reactfire.
reactfire 4.x obsoleted useAnalytics module, or have alternative way of useAnalytics module. I could not find the mention of useAnalytics.

**Reply by jhuleatt** [reply]
Hi @shiiinji, thank you for reporting this. I've filed it as issue https://github.com/FirebaseExtended/reactfire/issues/429 and included a workaround in the issue.

**Reply by shiiinji** [reply]
Thank you for your replying and share the work around of the issue.
